src/storage/mvcc/write.rs

```diff
@@ -70,16 +76,45 @@ impl std::fmt::Debug for Write {
                     .map(|v| hex::encode_upper(v))
                     .unwrap_or_else(|| "None".to_owned()),
             )
+            .field("protected", &self.protected)
             .finish()
     }
 }
 
 impl Write {
-    pub fn new(write_type: WriteType, start_ts: u64, short_value: Option<Value>) -> Write {
+    pub fn new(
```

Comment:
I prefer the field `protected` is false by default and `Write::new()` keeps its implementation. Add a `Write::new_rollback(start_ts: u64, protected: bool)` is sufficient.

Reply:
I am not sure if it's good to keep the original implementation of the `new` function. The advantage is that less code needs modifying but I prefer to make `protected` explicit.